packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/NamedTypeSymbolProvider.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.Generator.CSharp.Primitives;
+
+namespace Microsoft.Generator.CSharp.Providers
+{
+    public class NamedTypeSymbolProvider : TypeProvider
+    {
+        private INamedTypeSymbol _namedTypeSymbol;
+
+        public NamedTypeSymbolProvider(INamedTypeSymbol namedTypeSymbol)
+        {
+            _namedTypeSymbol = namedTypeSymbol;
+        }
+
+        public override string RelativeFilePath => throw new InvalidOperationException("This type should not be writting in generation");
+
+        public override string Name => _namedTypeSymbol.Name;
+
+        public override string Namespace => _namedTypeSymbol.ContainingNamespace.Name;
```

Comment:
this may not be correct - the namespaces in symbols have hierarchy, for example the symbol `System.Generator.CSharp.Primitives.NamedTypeSymbolProvider`, `ContainingNamespace.Name` should return `Primitives` if I did not remember it wrong.
There is a `GetDisplayString` method which allows you to get a full name from a symbol, we just need to pass in the `ContainingNamespace`

Reply:
Good catch, fixed and updated test to use nested namespace